lib/rwj_reporter.rb

```diff
@@ -0,0 +1,89 @@
+class RwjReporter
+  def self.print_reports_for_dates(start_date_str, end_date_str, log_file_dir=nil, output_dir=nil)
+    rr = RwjReporter.new(log_file_dir)
+    rr.channel_counts(start_date_str, end_date_str)
+    rr.print_channel_counts_files(output_dir)
+  end
+
+  def initialize(log_file_dir=nil)
+    @log_file_dir =
+      if log_file_dir
+        log_file_dir
+      else
+        log_file_dir_from_settings
+      end
+  end
+
+  def self.log_file_dir_from_settings
+    settings = YAML.load_file 'config/settings.yml'
+    settings['log_file_dir']
+  end
+
+  # expects a text file with xml that contains something like:
+  #  <body>1,1,21,unlimited,-1,, - Show Number 062<br />1,1,21,unlimited,-1,, - Show Number 196a<br /></body>
+  #
+  # where the <body> contains "lines" that end with show numbers (lines are separated by <br> tags, not necessarily line breaks).
+  # the show number should be the last thing on each line.  a show "number" might contain alpha chars, and is not strictly numeric.
+  #
+  # returns an array of show numbers from such a text file by finding matches for a simple regex in the text.
+  # so, in the above example, the result would be ["062", "196a"]
+  def self.get_show_numbers_from_log_file(fname)
+    log_file_content = open(fname).read
+    log_file_content.scan(/Show Number (\w+)</).flatten
+  end
+
+  def self.get_filenames_for_date_range(log_file_dir, start_date_str, end_date_str)
+    Dir.entries(log_file_dir).select do |fname|
+      get_date_str_from_filename(fname) >= start_date_str && get_date_str_from_filename(fname) <= end_date_str
+    end
+  end
+
+  # the following class methods can be considered private
+
+  def self.get_date_str_from_filename(fname)
+    fname_date_regex = /\.(\d{6})\-/
+    return '' unless fname.match(fname_date_regex)
+    fname.match(fname_date_regex)[1]
+  end
+
+  def self.add_shownum_to_channel_count(channel_count_hash, shownum)
+    channel_count_hash[shownum] =
+      if channel_count_hash.key? shownum
+        channel_count_hash[shownum] + 1
+      else
+        1
+      end
+  end
+
+  def channel_counts(start_date_str, end_date_str)
+    @start_date_str = start_date_str
+    @end_date_str = end_date_str
```

Comment:
This use of instance variables to set state feels odd to me.  It implies that `file_name` cannot be called until after `channel_counts` since that sets the state.

I don't know that I have a solid suggestion, but maybe make methods that depend on this state being set private since we don't want somebody inadvertently calling it from the outside w/o realizing the need to invoke the `channel_counts` method first.